### PR TITLE
Add v1.2 conformance report table

### LINKED ISSUE #3395: Conformance reports unavailable on the website for v1.1 and v1.2
**What would you like to be added**:

* https://gateway-api.sigs.k8s.io/implementations/v1.1/ does not show any conformance reports although the data exists https://github.com/kubernetes-sigs/gateway-api/tree/main/conformance/reports/v1.1.0 . The [link for v1.0](https://gateway-api.sigs.k8s.io/implementations/v1.0/) works fine

* v1.2 was recently released, so a section for this release should also be available on the website. Cilium has already uploaded a conformance report https://github.com/kubernetes-sigs/gateway-api/tree/main/conformance/reports/v1.2.0

**Why this is needed**:

### PR DESCRIPTION
**Pull request by xtineskim**
**What type of PR is this?**
/area conformance
/kind documentation

**What this PR does / why we need it**:
With 3 implementations conformance reports uploaded for v1.2, updating the table
**Which issue(s) this PR fixes**:

Fixes #3395

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by arkodg** [comment]
hey @xtineskim thanks for fixing this !
can we also get Kong's results in (https://github.com/kubernetes-sigs/gateway-api/pull/3426) ?

File: site-src/implementations/v1.1.md [hunk 1]
```diff
@@ -16,20 +16,21 @@ Implementations only appear in this page if they pass Core conformance for the r
...
+| GKE            | gke-gateway                   | 1.30.3-gke.1211000 | :white_check_mark: | :white_check_mark:     | :white_check_mark:      | :white_check_mark:       | :white_check_mark:                    | :white_check_mark:        | :x:                       | :x:                    | :x:                           | :x:                      | :x:                            | :x:                      | :x:                           | :x:                                 | :x:                                         | :x:                       | :x:                     | :x:                               | :x:                       |
```
Comment (arkodg) [inline site-src/implementations/v1.1.md:21]: seeing 3 identical entries for GKE's `1.30.3-gke.1211000` version

Comment (robscott) [reply site-src/implementations/v1.1.md:21]: Good catch - I think we should probably include "mode" from the conformance reports in this table. In this case, GKE is reporting separately for 3 different underlying LBs:

https://github.com/kubernetes-sigs/gateway-api/blob/f5262011c93037166cf967c40c0e2651fa15d842/conformance/reports/v1.1.0/gke-gateway/standard-1.30.3-gxlb-report.yaml#L13

**Review by youngnick — COMMENTED** [review]
LGTM once we add `mode` as @robscott suggested.

File: site-src/implementations/v1.1.md (hunk 1 above)
Comment (xtineskim) [reply site-src/implementations/v1.1.md]: good catch - didnt realize this was in the conformance reports. cmiiw, but this appears only in v1.1 and later? I changed the generate report script to reflect that!

**Review by arkodg — APPROVED** [review]
looks good, thanks for getting this in before Kubecon 😄 

/approve
/lgtm

**Comment by robscott** [comment]
Thanks @xtineskim!

/lgtm

**Pull request merged**